base/process.jl

```diff
@@ -72,6 +72,20 @@ backticks, e.g.
 """
 Cmd
 
+function hash(x::Cmd)
+    result = hash(x.exec)
+    result = result * 31 + hash(x.env)
+    result = result * 31 + hash(x.flags)
+    result = result * 31 + hash(x.dir)
+    result * 31 + hash(x.ignorestatus)
+end
```

Comment:
What's up with this idiom of multiplying by 31 and adding the hash? There are a number of examples of how to chain two-argument hashing over a series of values.